internal/api/toolbox/list.go

```diff
@@ -81,12 +85,15 @@ func (t *toolbox[T, K, V]) listWhenPermissionIsActivated(ctx echo.Context, param
 		return t.metadataOrFullList(persesContext, parameters, q)
 	}
 
-	result := make([]any, 0)
+	result := make([]any, 0, len(projects))
+	asynchronousRequests := make([]async.Future[any], 0, len(projects))
 	for _, project := range projects {
-		parameters.Project = project
-		listResult, listErr := t.metadataOrFullList(persesContext, parameters, q)
-		if listErr != nil {
-			return nil, listErr
+		asynchronousRequests = append(asynchronousRequests, async.Async(t.asyncMetadataOrFullList(persesContext, parameters, project, q)))
```

Comment:
Did you check that it works well? I think it's typically the kind of code that was not working before Go 1.22 - after the loop is over `project` would have been equal to the latest iterated one. Now I guess it works like this, but just to be sure.

Reply:
yeah I checked and I didn't get any issue !